src/DynamicData/Cache/ObservableCacheEx.BindSort.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2011-2023 Roland Pheasant. All rights reserved.
+// Roland Pheasant licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+using System.Collections.ObjectModel;
+using DynamicData.Binding;
+
+namespace DynamicData;
+
+/// <summary>
+/// ObservableCache extensions for BindAndSort.
+/// </summary>
+public static partial class ObservableCacheEx
+{
+    /// <summary>
+    /// Bind sorted data to the specified readonly observable collection.
+    /// </summary>
+    /// <typeparam name="TObject">The type of the object.</typeparam>
+    /// <typeparam name="TKey">The type of the key.</typeparam>
+    /// <param name="source">The source.</param>
+    /// <param name="targetList">The list to bind to.</param>
+    /// <param name="comparer">The comparer to order the resulting dataset.</param>
+    /// <returns>An observable which will emit change sets.</returns>
+    public static IObservable<IChangeSet<TObject, TKey>> BindAndSort<TObject, TKey>(
+        this IObservable<IChangeSet<TObject, TKey>> source,
+        IList<TObject> targetList,
+        IComparer<TObject> comparer)
```

Comment:
I would have `comparer` on all of these overloads be rolled into `BindAndSortOptions`, with the default being `null` and resolved to `Comparer<T>.Default` within the operator.

Reply:
Surely that would throw if the target object does not implement IComparable.  I think this would confuse most people and end up with a load of support issues being raised.

Reply:
It would indeed throw, yes. But it follows the pattern established by Microsoft in `System` namespaces. `List<T>.Sort()`, `Array.Sort()`, and `IEnumerable<T>.OrderBy()` all support being called without an `IComparer<T>` being given. If Microsoft thinks it's a reasonable compromise, that's good enough for me.

On a separate note, there's probably also room for an overload that takes a `Comparison<T>` , instead of an `IComparer<T>` which is also something all of the above system APIs have.

Reply:
Let's don that for Dynamic Data 2.  I do not mind making various changes for this version, but I think a change of concepts and limitations is part of the re-design